### PR TITLE
DBAL v3 compatibility

### PR DESCRIPTION
**Pull request by goetas**
|      Q       |   A
|------------- | -----------
| Type         | feature
| BC Break     | no
| Fixed issues | https://github.com/doctrine/migrations/issues/1141#issuecomment-919979808

### CODE REVIEW

**Comment by goetas** [comment]
blocked by https://github.com/doctrine/dbal/pull/4800

**Comment by dpfaffenbauer** [comment]
https://github.com/doctrine/dbal/pull/4811 fixed it right? anything I can do to help here? @goetas

**Comment by goetas** [comment]
Does anyone know a better way to test the dbal v3 and v2 with phpstan without duplicating the phpstan config file?

**Comment by mbabker** [comment]
> Does anyone know a better way to test the dbal v3 and v2 with phpstan without duplicating the phpstan config file?

Take a look at how it's handled in the ORM.

**Comment by derrabus** [comment]
CI has failed because getcomposer.org was down this morning. I've restarted it.

**Review by derrabus — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "require-dev": {
         "ext-pdo_sqlite": "*",
         "doctrine/coding-standard": "^8.0",
-        "doctrine/orm": "^2.6",
+        "doctrine/orm": "^2.6@dev",
```
Comment (greg0ire) [inline composer.json]: This is no longer needed, is it?

**Comment by greg0ire** [comment]
Should we squash everything or are there commits in there you would like to keep independent from the rest?

**Review by greg0ire — APPROVED** [review]

**Comment by brusch** [comment]
Thanks a lot guys! 
@goetas is there any eta for the `3.3.0` release?

**Pull request merged**